docs/dev/docker.rst

```diff
@@ -0,0 +1,51 @@
+.. _docker:
+
+Builds and Dependency Management
+================================
+
+General structure of Programming Exercise Execution
+---------------------------------------------------
+
+Artemis uses Docker containers for running programming exercises. This ensures that the code of the students doesn't have direct access to the hardware of the build agents.
+To reduce the needed time of each test run, these docker containers already contain often used dependencies like JUnit.
+
+There are currently docker containers for:
+
+- Maven (Java and Kotlin): https://github.com/ls1intum/artemis-maven-docker
+- OCaml: https://github.com/ls1intum/artemis-ocaml-docker
+- Python: https://github.com/ls1intum/artemis-python-docker
+- C: https://github.com/ls1intum/artemis-c-docker
+- Haskell (external): https://github.com/lukasstevens/docker-fpv-stack
+- Swift (external): https://github.com/norio-nomura/docker-swiftlint
+- Assembler (external): https://hub.docker.com/r/tizianleonhardt/era-artemis-assembler
+- VHDL (external): https://hub.docker.com/r/tizianleonhardt/era-artemis-vhdl
+
+Steps for Updating the DockerBuilds used in Artemis
+---------------------------------------------------
+Updating can only be done with access to the account on Dockerhub. If you need access, contact `Stephan Krusche <krusche@in.tum.de>`_.
+
+1. Update the dependencies via a pull request in the docker. Each docker has its own repository as listed above.
```

Comment:
I am not sure, does "each docker" mean each "Docker container"? I would find it easier if  "Docker container" is used throughout the document as in the section above.

Reply:
I would remove the '.' at the end of the line to be consistent with the other steps

Reply:
I think this is about Docker Images.